### PR TITLE
Area/multiple devices and name support for thingtalk automations

### PR DESCRIPTION
**Pull request by bramkragten**

### CODE REVIEW

File: src/components/device/ha-area-devices-picker.ts [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    }
+    const areas = this._getDevices(
+      this._devices,
+      this._areas,
+      this._entities,
+      this.includeDomains,
+      this.excludeDomains,
+      this.includeDeviceClasses
+    );
+    if (!this._areaPicker || areas.length === 0) {
```
Comment (balloob) [inline src/components/device/ha-area-devices-picker.ts]: Isn't this weird? Shouldn't this be taken care of by whoever renders this component ?

File: src/panels/config/automation/thingtalk/ha-thingtalk-placeholders.ts [hunk 2]
```diff
@@ -189,17 +244,72 @@ export class ThingTalkPlaceholders extends SubscribeMixin(LitElement) {
...
+    }
+    const foundDevice = this._devices.find((device) => device.id === deviceId);
+    if (!foundDevice) {
+      return "";
+    }
+    return foundDevice.name_by_user || foundDevice.name || "";
+  }
+
+  private _searchNames() {
+    if (this._search && this._areas && this._devices) {
```
Comment (balloob) [inline src/panels/config/automation/thingtalk/ha-thingtalk-placeholders.ts]: Use guard clause. 

```python
    if (!this._search || !this._areas || !this._devices) { return }
```

File: src/panels/config/automation/thingtalk/ha-thingtalk-placeholders.ts [hunk 3]
```diff
@@ -189,17 +244,72 @@ export class ThingTalkPlaceholders extends SubscribeMixin(LitElement) {
...
+      this._search = false;
+      Object.entries(this.placeholders).forEach(([type, placeholders]) =>
+        placeholders.forEach((placeholder) => {
+          if (placeholder.name) {
```
Comment (balloob) [inline src/panels/config/automation/thingtalk/ha-thingtalk-placeholders.ts]: Guard clause

**Comment by balloob** [comment]
This is wild and amazing.

File: src/components/device/ha-area-devices-picker.ts (hunk 1 above)
Comment (bramkragten) [reply src/components/device/ha-area-devices-picker.ts]: You want to split the area and device picker in the parent? The parent should also know if there are areas that meet the requirements then...

Comment (balloob) [reply src/components/device/ha-area-devices-picker.ts]: hmmmmmm okay let's stick with this.

**Review by balloob — APPROVED** [review]

**Pull request merged**